proposals/on-demand-definitions.md

```diff
@@ -0,0 +1,733 @@
+# On-Demand Definitions Protocol
+
+A protocol for defining custom elements on-demand without relying on top-level
+side-effects.
+
+**Author**: [dgp1130](https://github.com/dgp1130/)
+
+**Status**: Draft
```

Comment:
Administrivia, not specifically making any statement on the contents of the PR/proposal...

Can you set this a "Proposal" and add this to the list in https://github.com/webcomponents-cg/community-protocols?tab=readme-ov-file#proposals.

Reply:
Oh, the template says "Draft", so I was assuming that's the first phase for a protocol. Updated this to "Proposal" and also updated the template to match, let me know if the template was intended to be "Draft".